crates/sui-core/src/authority_active/checkpoint_driver/mod.rs

```diff
@@ -507,31 +508,53 @@ where
         .unwrap_or(0);
 
     for seq in full_sync_start..latest_known_checkpoint.summary.sequence_number {
-        debug!("Full Sync ({name:?}): {seq:?}");
+        debug!(name = ?state.name, ?seq, "Full Sync",);
         let (past, contents) =
-            get_one_checkpoint(net.clone(), seq, true, &available_authorities).await?;
+            get_one_checkpoint_with_contents(net.clone(), seq, &available_authorities).await?;
 
-        if let Err(err) =
-            checkpoint_db
-                .lock()
-                .process_checkpoint_certificate(&past, &contents, &net.committee)
-        {
+        sync_checkpoint_certs(
```

Comment:
Question: Do we want to use a synchronous process here, or do we want to rely on the pending execution system and wait for it to catch up, as @lxfind started doing here: https://github.com/MystenLabs/sui/pull/2857/files#r913120773

Reply:
Worth a note that currently using node sync mechanism to sync checkpoint transactions may not work, as there is no guarantee that a shared object lock won't be set twice

Reply:
I have a plan for addressing this, started a slack convo about it. can talk tomorrow when you're back.